dbt/include/global_project/macros/adapters/show.sql

```diff
@@ -1,22 +1,26 @@
+{#
+    We expect a syntax error if dbt show is invoked both with a --limit flag to show
+    and with a limit predicate embedded in its inline query. No special handling is
+    provided out-of-box.
+#}
 {% macro get_show_sql(compiled_code, sql_header, limit) -%}
-  {%- if sql_header -%}
+  {%- if sql_header is not none -%}
   {{ sql_header }}
-  {%- endif -%}
-  {%- if limit is not none -%}
+  {%- endif %}
   {{ get_limit_subquery_sql(compiled_code, limit) }}
-  {%- else -%}
-  {{ compiled_code }}
-  {%- endif -%}
 {% endmacro %}
 
-{% macro get_limit_subquery_sql(sql, limit) %}
-  {{ adapter.dispatch('get_limit_subquery_sql', 'dbt')(sql, limit) }}
-{% endmacro %}
+{#
+    Not necessarily a true subquery anymore. Now, merely a query subordinate
+    to the calling macro.
+#}
+{%- macro get_limit_subquery_sql(sql, limit) -%}
+  {{ adapter.dispatch('get_limit_sql', 'dbt')(sql, limit) }}
+{%- endmacro -%}
 
-{% macro default__get_limit_subquery_sql(sql, limit) %}
-    select *
-    from (
-        {{ sql }}
-    ) as model_limit_subq
-    limit {{ limit }}
+{% macro default__get_limit_sql(sql, limit) %}
+  {{ compiled_code }}
```

Comment:
@VersusFacit and @mikealfare  Looks like this line should be using the function parameter `{{ sql }}`. It works with `{{ compiled_code }}` I assume because of the [context being set in the get_show_sql macro](https://github.com/dbt-labs/dbt-core/blob/1d3d315249c81dbca70452e5a14fbfd0edad6b2a/core/dbt/task/show.py#L34), but that's likely hiding a future bug.

I could be wrong, but ran into this when providing a custom version of get_show_sql in my project.